tox/venv.py

```diff
@@ -397,25 +402,51 @@ def tox_testenv_create(venv, action):
     # if self.getcommandpath("activate").dirpath().check():
     #    return
     config_interpreter = venv.getsupportedinterpreter()
-    args = [sys.executable, '-m', 'virtualenv']
-    if venv.envconfig.sitepackages:
-        args.append('--system-site-packages')
-    # add interpreter explicitly, to prevent using
-    # default (virtualenv.ini)
-    args.extend(['--python', str(config_interpreter)])
+    venv.session.make_emptydir(venv.path)
+    basepath = venv.path.dirpath()
+    basepath.ensure(dir=1)
+    # FIXME there should be a better way to get if we're not using the default
+    default = EnvcreatecmdOption.default.replace('{envdir}', str(venv.path)).split()
+    if venv.envconfig.envcreate_command != default:
+        args = venv.envconfig.envcreate_command
+        i = args.index('{basepython}')
+        args[i] = str(config_interpreter)
+        # i = args.index('{envdir}')
+        # args[i] = venv.path.basename
+        # FIXME do this in config.py
+        # currently checking it here since I don't know how to use {envdir}
+        # correctly
+        if venv.path not in args:
+            raise tox.exception.ConfigError(
+                "'envcreate_command' must contain '{envdir}' substitution")
+    else:
+        args = [sys.executable, '-m', 'virtualenv']
+        if venv.envconfig.sitepackages:
+            args.append('--system-site-packages')
+        # add interpreter explicitly, to prevent using
+        # default (virtualenv.ini)
+        args.extend(['--python', str(config_interpreter)])
+        args.append(venv.path.basename)
+
     # if sys.platform == "win32":
     #    f, path, _ = py.std.imp.find_module("virtualenv")
     #    f.close()
     #    args[:1] = [str(config_interpreter), str(path)]
     # else:
-    venv.session.make_emptydir(venv.path)
-    basepath = venv.path.dirpath()
-    basepath.ensure(dir=1)
-    args.append(venv.path.basename)
     venv._pcall(args, venv=False, action=action, cwd=basepath)
 
 
 @hookimpl
+def tox_testenv_prepare(venv, action):
+    # FIXME this is probably not the right way to do things
```

Comment:
Doing it this way also runs it without a shell, so things like `curl a | b` don't work.
Also, we're going through the whitelist, which makes it annoying to use.